unity-sdk/AndroidPlugin/OptimoveUnityPlugin/src/main/java/com/optimove/unity/plugin/UnityProxy.java

```diff
@@ -1,4 +1,376 @@
 package com.optimove.unity.plugin;
 
+import android.content.Context;
+
+import com.optimove.android.Optimove;
+import com.optimove.android.optimobile.InAppDeepLinkHandlerInterface;
+import com.optimove.android.optimobile.InAppInboxItem;
+import com.optimove.android.optimobile.InAppInboxSummary;
+import com.optimove.android.optimobile.OptimoveInApp;
+import com.unity3d.player.UnityPlayer;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.net.URL;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.TimeZone;
+
 public class UnityProxy {
+    private static JSONObject pendingPush;
+
+    public static void setUserId(String userId) {
+        try {
+            Optimove.getInstance().setUserId(userId);
+        } catch (Exception e) {
```

Comment:
just noting a difference, in ios we dont catch any exceptions (same in cordova)

Reply:
is there any reason to catch an exception here? the method setUserId of the SDK doesn't throw any.

Reply:
i think it was copied from cordova, and who was doing user association in android cordova? ;) 

i would even remove

Reply:
lol, by that I can tell that I'm wiser now, I will remove it.